Prep for v2.1 release.

ronrat opened the pull request:
- Updated release notes in the Nuget configuration to include changes that were made since we started prepping v2.1.
- TinyIoCSinglePrivateCtorSpec compile was failing in the release build configuration because it referenced internal class TinyIoCResolutionException in the SpecEasy project, which is only visible in debug build. The TinyIoCSinglePrivateCtorSpec is now only included in the debug build.
- Specs project build was failing in the .NET 4.5 build configuration because the ExternalLib project was targeting .NET 4.5.1.
- update Nuget build to use MSBuild v14.0.

As soon as this PR is approved, I will push the new release to Nuget.

rockhymas commented:
Overall this looks good. An alternative to the #if DEBUG around TinyIoCSinglePrivateCtorSpec  would be to make the speceasy.specs assembly only build in debug configurations. Obviously there's a tradeoff there - being able to run tests in release configs vs having to exclude tests explicitly like this (and remember to run tests in both configs, just in case). I could be argued either way, I think, but if it were my call, I'd probably just build speceasy.specs in debug configs.

dbertram commented:
It looks like the copy of TinyIoC we're using has already drifted quite a bit from upstream. Would it be easier to make `TinyIoCResolutionException` public and make it's constructors internal?

Then you still can't construct that class outside of SpecEasy, but the tests will build/run/pass regardless of build configuration.

rockhymas commented:
Hm, I'd rather not make it public. Maybe instead of testing TinyIoC directly, we should recognize that it is implementation details, and rewrite the tests that exercise TinyIoC behavior to do it through the standard SpecEasy API. In the process we would want to catch any TinyIoC specific exceptions and wrap them in SpecEasy exceptions. Either way, I think the changes in the PR are fine to get us unblocked on updating the release. I think we can tackle this cleanup separately.

dbertram commented:
@ronrat How do you feel about you/us getting this branch updated/rebased onto the tip of master and then releasing a "SpecEasy 2.1" to NuGet?

Then all the new hotness that's been added recently will be bundled up and available to anyone who wants to continue targeting NUnit 2.x before we do the major version bump to support NUnit 3.x.

If there's stuff I can help out with, please let me know!

ronrat commented:
Definitely. I can carve out some time to do that. 

One thing that I haven’t mentioned is that the new release doesn’t actually work in one of our TrackAbout solutions. We’re encountering an issue with our database integration tests because the setup and teardown isn’t working as expected. My current suspicion is that it has to do with the async changes that were made. I believe the setup and teardown are happening on different threads and, since we’re using NH sessions per thread, the teardown is trying to use another session. I’ve spent some time digging into it but haven’t gotten to the bottom of yet. Anyway, that’s one of the things that distracted me from getting this v2.1 change completed.

Ron

On Thu, Jul 14, 2016 at 12:46 PM Dane Bertram

<
mailto:Dane Bertram <notifications@github.com>
> wrote:

https://github.com/ronrat
How do you feel about you/us getting this branch updated/rebased onto the tip of master and then releasing a "SpecEasy 2.1" to NuGet?

Then all the new hotness that's been added recently will be bundled up and available to anyone who wants to continue targeting NUnit 2.x before we do the major version bump to support NUnit 3.x.

If there's stuff I can help out with, please let me know!

—

You are receiving this because you were mentioned.

Reply to this email directly,
https://github.com/speceasy/speceasy/pull/36#issuecomment-232756209
, or
https://github.com/notifications/unsubscribe/AAuATDA-rsECzjrLoGTxuUuvFqflp57mks5qVoPxgaJpZM4G0YCI
.

https://github.com/speceasy/speceasy/pull/36#issuecomment-232756209

dbertram commented:
@ronrat Interesting re: async causing problems with some of your tests. If it's not a giant pain to get a minimal repro up somewhere, I'd be happy to take a crack at it with you.

ronrat commented:
@dbertram I think this PR is ready to go. Added a few more changes for your (or anyone's) perusal. When this is approved and merged, I can push the v2.1 release to NuGet.

dbertram commented:
@ronrat Yeah, this PR looks good to me.

What do you think about getting https://github.com/speceasy/speceasy/pull/33 and possibly https://github.com/speceasy/speceasy/pull/46 merged in before we drop 2.1?

We can do those as a subsequent release though, if you'd rather.

ronrat commented:
@dbertram I'm happy to wait for those other PRs. I would be nice to put this PR to bed soon.

ronrat commented:
@dbertram, @appakz, et al. Happy Thanksgiving! 🦃 After having been away from writing a lot of unit tests lately, I ran into a situation where it would have been helpful to have the SUT build override feature of #33. That got me to thinking about that PR and #46. Any chance we could pick back up on those soon?

dbertram commented:
@ronrat Happy (belated) 🦃 day!

I'll talk to @rockhymas and see if we can settle on what to do with #46, and then we can get v2.1 out the door.

dbertram commented:
@ronrat Other than rebasing this, I think we're ready to :shipit:!

Down the road I'd like to talk to you about setting up an automated AppVeyor build + deploy to NuGet pipeline to streamline this stuff. But I'm anxious to get this big batch of improvements out there first. :)

ronrat commented:
@dbertram Super!

I'd love to work with you on an AppVeyor build. I was actually looking at AppVeyor the other day for another project.

dbertram commented:
@ronrat I think this just needs to be rebased and have the copyright updated to 2017.

I started working on the appveyor stuff, but I ran into some issues with the 4.5 vs 4.5.1 build configs; https://ci.appveyor.com/project/dbertram/speceasy/build/job/eg6d9t5ruu9en3na. Sometime when you're free maybe we could sync up to get that working.

Speaking of appveyor, do you manage the TrackAbout nuget account? I created a SpecEasy nuget account, but we'll need whoever manages the TrackAbout user to add the new SpecEasy user as a co-owner for the SpecEasy nuget package.

jsternal commented:
@dbertram I can coordinate adding the new SpecEasy user for the nuget package.

ronrat commented:
@dbertram I no longer work for TrackAbout. Thanks @jsternal for taking up the nuget account work.

I will look at rebasing and merging in the next week.

dbertram commented:
@jsternal Sweet! Thank you!

Related: Once we get publishing to nuget.org automated via appveyor, unless folks are opposed I'd really like to make SpecEasy a strongly-named assembly. Currently we maintain a fork of SpecEasy internally whose only difference is a different name (to avoid confusion) + assembly signing b/c any assemblies we use must be strongly-named.

I'll file a separate github issue about the automation + signing, but I wanted to mention it here since we're talking about releases anyway.

ronrat commented:
@dbertram I've rebased this and added a [small commit](https://github.com/speceasy/speceasy/commit/094109a6b6e3f3aa8fa990f9a6d8321ee0e5251b). I think this is ready to be merged. Once it's merged, whomever has the nuget api key can [follow the instructions](https://github.com/speceasy/speceasy/tree/rlr/prep-2.1-release/build) to push to nuget.

ronrat commented:
@dbertram I just included [a commit](https://github.com/speceasy/speceasy/pull/36/commits/67a9782dc07cfa6172ee04e0863b272f0122d9bb) that fixes the appveyor builds. The problem was that the nuspec file explicitly specified that it should included the release 4.5 bin in the package but that would only exist when the release 4.5 build configuration was run. The nuspec was built assuming that both release builds would be run before the nuget package is built. That's what the `[build.nuget.bat](https://github.com/speceasy/speceasy/blob/094109a6b6e3f3aa8fa990f9a6d8321ee0e5251b/build/build.nuget.bat)` file does. To build one nuget packages with both the 4.5 and 4.5.1 versions, we will probably need another job that runs both builds and then builds the nuget package.

dbertram commented:
@ronrat This lgtm.

I'm still super new to appveyor, so we'll have to figure out the best we to (hopefully) get to fully automated build + test + generate/deploy nuget package. My guess is we'll leave things as-is (for now) in terms of fully automatic build + test for PR builds, and then create a manually run appveyor build that does the build + test + generate and deploy nuget package.

@jsternal Any news on getting the [SpecEasy](https://www.nuget.org/profiles/SpecEasy) nuget profile added as a co-owner to the SpecEasy nuget package? If it's useful, here are the [nuget docs on adding additional package owners](http://blog.nuget.org/20130205/managing-package-owners.html).

jsternal commented:
@dbertram working on it - I don't have direct access to the account, so I'm working by remote control here. Should be soon.

ronrat commented:
Am I supposed to merge my own PR or are we waiting on something else to get this thing merged?

dbertram commented:
@ronrat I view you as one of the maintainers, so I assumed you would merge.
